libp2p/dialer.nim

```diff
@@ -169,47 +173,60 @@ proc internalConnect(
     forceDial: bool,
     reuseConnection = true,
     dir = Direction.Out,
-): Future[Muxer] {.async.} =
+): Future[Muxer] {.async: (raises: [DialFailedError, CancelledError]).} =
   if Opt.some(self.localPeerId) == peerId:
-    raise newException(CatchableError, "can't dial self!")
+    raise newException(DialFailedError, "can't dial self!")
 
   # Ensure there's only one in-flight attempt per peer
   let lock = self.dialLock.mgetOrPut(peerId.get(default(PeerId)), newAsyncLock())
-  try:
-    await lock.acquire()
+  await lock.acquire()
+  defer:
+    if lock.locked():
+      try:
+        lock.release()
+      except:
+        raiseAssert "checked with if"
```

Comment:
Why is this `if` necessary?
if `await lock.acquire()` is succesful, then `lock.locked()` should return `true` unless i'm missing something.

Reply:
no, it is not necessary